frontend/src/scenes/retention/RetentionTable.tsx

```diff
@@ -92,28 +48,65 @@ export function RetentionTable({ inCardView = false }: { inCardView?: boolean })
         loadMorePeople()
     }
 
+    const headings = ['Cohort', 'Size', ...results.map((x) => x.label)]
+
+    const rows = Array.from(Array(maxIntervalsCount).keys()).map((rowIndex: number) => [
```

Comment:
non-blocking... there's a lot of logic in here... if at least some was in the logic we could test it

Reply:
re-reading this... it's just the word logic over and over again

I'll tell myself off
![apologise-to-god](https://user-images.githubusercontent.com/984817/188485555-23e4200e-0547-407b-8dd8-51dc70c59b0a.jpeg)

some of this could be in the retentionTableLogic 🤣

Reply:
Definitely agree. I tried to touch as little "other" code as I could but this could be worth the extra code diff

Reply:
Just fyi @pauldambra I moved some stuff to the logic and added tests - good call